var/spack/repos/builtin/packages/hypre/package.py

```diff
@@ -109,3 +109,14 @@ def install(self, spec, prefix):
                 sstruct('-in', 'test/sstruct.in.default', '-solver', '40',
                         '-rhsone')
             make("install")
+
+    @property
+    def libs(self):
+        libraries = ['libHYPRE']
+        liblist = (self.spec['lapack'].libs + self.spec['blas'].libs)
```

Comment:
For consistency with other, do not add dependencies (lapack) here. This should be done outside per-package basis.

Ps. Sorry, I was not clear enough in what I suggested

Reply:
No worries! It's an easy enough change.

Reply:
@davydden, I'm not sure I understand your objection to adding the lapack and blas libs to hypre's libs - who's going to add them to hypre's libs? Or, do you expect every package that depends on hypre to add them?

I'm asking because I've been looking into adding the libs property to most of mfem's dependencies (I am an mfem developer) and my solution (not pushed in any branch yet) was to include the lapack dependencies to hypre.

Reply:
@v-dobrev Whoever is packaging software that depends on hypre will also have to get flags from the LAPACK and BLAS libraries used in building hypre. I think the reasoning here is that the design contract for the `libs` method is that it should only return relevant flags for that particular package. This design means that packagers will need to know about transitive dependencies, like hypre's dependency on LAPACK and BLAS -- which is annoying -- but the tradeoff is that `libs` doesn't do anything unexpected like overlinking, so it's easier to reason about.

My first instinct was to dislike the design decision, but I think its virtue (in addition to consistency) is that it's usually a lot easier to spot underlinking a library than overlinking it, so I appreciate the design more now.

Reply:
If that is really the design decision, I'm really puzzled - it makes the libs property pretty much useless. How do I know how a library I depend on was build and what exactly do I need to link with? Many packages will have complicated logic about what they need to link with based on their variants - so all that logic will have to be duplicated in all dependent packages? I don't see how that makes sense.

Reply:
The reason is as @goxberry explained.

Reply:
OK. Just to confirm: is the same behavior is expected from the `.headers` property?

Thanks for the quick responses.

Reply:
Yes